### ISSUE TITLE
Translations file scanning should not go to vendor folder

### ISSUE THREAD

**Issue by tswfi** [PrestaShop/PrestaShop#24987]
#### Describe the bug

If a module has a vendor folder with bit more stuff the translation view is really slow and might either timeout or run out of memory.

The translation scanning in a module should not look into `vendor` folder (and probably `lib` and some others too).

#### Expected behavior

Vendor folder should not be scanned

#### Steps to Reproduce

Create minimal module with 

```
    public function isUsingNewTranslationSystem()
    {
        return true;
    }
```

Go to bo module translation for that module
=> the translation loads nicely

Now add a composer.json with some depencies (tested with `iio/libmergepdf`) and run

```
composer install --no-dev
```

go back to the module translation page
=> the page will either be really slow, timeout or run out of memory

#### Additional information

Some slack discussion about this:
![image](https://user-images.githubusercontent.com/1352132/122231516-2931b480-cec3-11eb-9975-38286dcba953.png)

* PrestaShop version: develop branch
* PHP version: (whatever comes from docker-compose up in develop branch right now :)

**Comment by tswfi** [comment]
and a simple module from module generator for testing this
https://github.com/tswfi/translationtesting

**Comment by hibatallahAouadni** [comment]
Hello @tswfi 

Thanks for the detailed issue and the test module, I reproduce the issue with develop, PS1.7.6.7 & PS1.7.7.5, see the attached screen record:

https://drive.google.com/file/d/1VP2GXOrajjllWTYiDgBBeQ7Dv4Qti572/view

Without running this command: `composer install --no-dev` the translation loaded quickly but after executing it it takes over 20 minutes to display an error **Internal Server Error**. I’ll add this to the debug backlog so that it’s fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved.

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by presteamshop** [comment]
Hi everyone,

A solution to this issue is necessary, be it a patch or something temporary, since it is impossible to use the new translation tool in the modules when working with "composer" and we have the "vendor" folder.

![image](https://user-images.githubusercontent.com/48451631/137045346-d41cb865-e747-4d77-bfdc-5587af777982.png)

The problem is still in 1.7.8.0:
![image](https://user-images.githubusercontent.com/48451631/137045627-4250b430-67d2-4c39-bc4c-998764574e0f.png)


Best regards.

**Comment by tswfi** [comment]
I got this far by following the request:
https://github.com/PrestaShop/PrestaShop/blob/98d0ec6dbc32afdf1c7bd1351a478e138114cd09/src/Core/Translation/Builder/TranslationsTreeBuilder.php#L100

I wonder if there is some kind of flowchart on what happens when searching for the strings somewhere... would help debugging this.

Poking around I found this:

https://github.com/PrestaShop/TranslationToolsBundle/blob/66c85b48c5052418f34e32e34e3e0b653b94b367/Translation/Extractor/PhpExtractor.php#L164 

it uses the Symfony finder.. adding ->exclude('vendor') is a ugly hack but fixes this..

```php
    protected function extractFromDirectory($directory)
    {
        return $this->getFinder()->files()->name('*.php')->exclude('vendor')->in($directory);
    }
```

there is probably a better way but the whole translation extraction and handling is so convoluted and does not have enough comments to follow the process through that I give up here...

tl;dr Someone who knows how this thing works needs to take a look at it :)

**Comment by tswfi** [comment]
https://symfony.com/doc/current/components/finder.html

would probably be good to say: `->ignoreVCSIgnored(true);`

and `->notPath('vendor');` might be better than exclude?

and `->size('640k');`  ought to be enough for anybody ;)

**Comment by presteamshop** [comment]
Hello @tswfi 

Thank you very much for answering so soon and with a temporary solution, it was very necessary to be able to advance temporarily, while PrestaShop can provide a definitive solution, because until they do this, we cannot enable the new form of translation in the modules that we use Composer.

Solution:
![image](https://user-images.githubusercontent.com/48451631/137215036-63bb1c57-c465-48d4-a922-a900f86030c3.png)

Result: 
![image](https://user-images.githubusercontent.com/48451631/137215046-b01598fc-fd2b-4e4a-bddd-65d70f16cd04.png)

Best regards.

**Comment by presteamshop** [comment]
@eternoendless I think it is something important to verify and give a prompt solution.
